src/main/kotlin/no/nav/syfo/cronjob/statusendring/PublishDialogmoteStatusEndringService.kt

```diff
@@ -0,0 +1,69 @@
+package no.nav.syfo.cronjob.statusendring
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.dialogmote.avro.KDialogmoteStatusEndring
+import no.nav.syfo.dialogmote.database.*
+import no.nav.syfo.dialogmote.database.domain.*
+import no.nav.syfo.dialogmote.domain.*
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.domain.Virksomhetsnummer
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+
+class PublishDialogmoteStatusEndringService(
+    private val database: DatabaseInterface,
+    private val dialogmoteStatusEndringProducer: DialogmoteStatusEndringProducer,
+) {
+    fun getDialogmoteStatuEndretToPublishList(): List<DialogmoteStatusEndret> {
+        return database.getMoteStatusEndretNotPublished().map {
+            it.toDialogmoteStatusEndret()
+        }
+    }
+
+    fun publishAndUpdateDialogmoteStatusEndring(
+        dialogmoteStatusEndret: DialogmoteStatusEndret,
+    ) {
+        val moteId = dialogmoteStatusEndret.moteId
+
+        val kDialogmoteStatusEndring = createKDialogmoteStatusEndring(
+            dialogmoteStatusEndret = dialogmoteStatusEndret,
+            dialogmoteTidStedList = database.getTidSted(moteId).map {
```

Comment:
Mange database-kall på rad her. Kanskje det hadde vært bedre ha disse metodene på Connection, og så gjøre connection.use?

Reply:
Hm, nei det blir nok en for stor endring..

Reply:
Skulle helst ha brukt DialogmoteService her, men det hadde blitt en del kluss med gjennbruk i tester uten et bibliotek for dependency-injection,